lib/src/compatKtLint0Dot48Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot48Dot0Adapter.java

```diff
@@ -0,0 +1,126 @@
+import com.pinterest.ktlint.core.api.EditorConfigDefaults;
+import com.pinterest.ktlint.core.api.EditorConfigOverride;
+import com.pinterest.ktlint.core.api.UsesEditorConfigProperties;
+import com.pinterest.ktlint.core.api.editorconfig.EditorConfigProperty;
+import com.pinterest.ktlint.ruleset.experimental.ExperimentalRuleSetProvider;
+import com.pinterest.ktlint.ruleset.standard.StandardRuleSetProvider;
+
+import kotlin.Pair;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function2;
+
+public class KtLintCompat0Dot48Dot0Adapter implements KtLintCompatAdapter {
+
+	static class FormatterCallback implements Function2<LintError, Boolean, Unit> {
+		@Override
+		public Unit invoke(LintError lint, Boolean corrected) {
+			if (!corrected) {
+				KtLintCompatReporting.report(lint.getLine(), lint.getCol(), lint.getRuleId(), lint.getDetail());
+			}
+			return null;
+		}
+	}
+
+	@Override
+	public String format(final String text, final String name, final boolean isScript,
+			final boolean useExperimental,
+			final Map<String, String> userData,
+			final Map<String, Object> editorConfigOverrideMap) {
+		final FormatterCallback formatterCallback = new FormatterCallback();
+
+		Set<RuleProvider> allRuleProviders = new LinkedHashSet<>(
+				new StandardRuleSetProvider().getRuleProviders());
+		if (useExperimental) {
+			allRuleProviders.addAll(new ExperimentalRuleSetProvider().getRuleProviders());
+		}
+
+		EditorConfigOverride editorConfigOverride;
+		if (editorConfigOverrideMap.isEmpty()) {
+			editorConfigOverride = EditorConfigOverride.Companion.getEmptyEditorConfigOverride();
+		} else {
+			editorConfigOverride = createEditorConfigOverride(allRuleProviders.stream().map(
+					RuleProvider::createNewRuleInstance).collect(
+							Collectors.toList()),
+					editorConfigOverrideMap);
+		}
+
+		return KtLint.INSTANCE.format(new KtLint.ExperimentalParams(
+				name,
+				text,
+				allRuleProviders,
+				userData,
+				formatterCallback,
+				isScript,
+				false,
+				EditorConfigDefaults.Companion.getEmptyEditorConfigDefaults(),
+				editorConfigOverride,
+				false));
+	}
+
+	/**
+	 * Create EditorConfigOverride from user provided parameters.
+	 */
+	private static EditorConfigOverride createEditorConfigOverride(final List<Rule> rules, Map<String, Object> editorConfigOverrideMap) {
+		// Get properties from rules in the rule sets
+		Stream<EditorConfigProperty<?>> ruleProperties = rules.stream()
+				.filter(rule -> rule instanceof UsesEditorConfigProperties)
+				.flatMap(rule -> ((UsesEditorConfigProperties) rule).getEditorConfigProperties().stream());
+
+		// get complete list of supported properties in DefaultEditorConfigProperties.INSTANCE
+		List<EditorConfigProperty<?>> editorConfigProperties = new ArrayList<>(DefaultEditorConfigProperties.INSTANCE.getEditorConfigProperties());
+		editorConfigProperties.add(DefaultEditorConfigProperties.INSTANCE.getKtlintDisabledRulesProperty());
```

Comment:
I think `ktlint_disabled_rules` has been added to `DefaultEditorConfigProperties` in 0.48.0, already, so this whole `new ArrayList` part would be useless now. I introduced it in https://github.com/diffplug/spotless/pull/1378 for a fix, but if it's in the `DefaultEditorConfigProperties` we can get rid of it.

Reply:
Thanks a ton @bcmedeiros for keeping an eye on this. @davidburstrom your call, I'm okay to merge it as-is, or if you want to make the change suggested above first I'm happy to wait.

Reply:
Yeah, it definitely doesn't hurt.

Reply:
I'll take a look at the suggested change.